offix-offline/src/main/java/org/aerogear/offixoffline/NetworkChangeReceiver.kt

```diff
@@ -16,11 +17,18 @@ class NetworkChangeReceiver : BroadcastReceiver() {
 
     override fun onReceive(context: Context?, intent: Intent?) {
         /*
-        Offload the work to an IntentService that syncs the mutations with the server
+        Offload the work to an IntentService that syncs the mutations with the server only when list is not empty.
          */
-        if (intent?.action == "android.net.conn.CONNECTIVITY_CHANGE") {
+        if (intent?.action == "android.net.conn.CONNECTIVITY_CHANGE" &&
+            OfflineList.getInstance().offlineArrayList.isNotEmpty()
+        ) {
             val foregroundService = Intent(context, OfflineSyncService::class.java)
             context?.startService(foregroundService)
+//            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+//                context?.startForegroundService(foregroundService)
+//            } else {
```

Comment:
Remove this instead of commenting.
It's a good practice not to leave unnecessary stuff in the production code.